### PR TITLE
Introduces DeepVAR model

### PR DESCRIPTION
**Pull request by mbohlkeschneider**
*Description of changes:*

Introduces the DeepVAR model for multivariate forecasts. This is the basic abstraction leading up to the GP Copula model (which comes in the next PR). 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/model/deepvar/_estimator.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        self.freq = freq
+
+    def __call__(self, index: pd.DatetimeIndex) -> np.ndarray:
+        values = getattr(index, self.freq)
+        num_values = max(values) + 1
+        steps = [x * 2.0 * np.pi / num_values for x in values]
+        return np.vstack([np.cos(steps), np.sin(steps)])
+
+
+def get_granularity(freq_str: str) -> Tuple[int, str]:
```
Comment (jaheba) [inline src/gluonts/model/deepvar/_estimator.py]: Don't we have this code elsewhere as well?

File: src/gluonts/model/deepvar/_network.py [hunk 2]
```diff
@@ -0,0 +1,822 @@
...
+            *lagged_values, num_args=len(indices), dim=1
+        ).transpose(axes=(0, 2, 3, 1))
+
+    def unroll(
+        self,
+        F,
+        lags: Tensor,
+        scale: Tensor,
+        time_feat: Tensor,
+        target_dimensions: Tensor,
```
Comment (lcallot) [inline src/gluonts/model/deepvar/_network.py]: It's kind of confusing that target_dim and target_dimension refer to two fairly different things.

Comment (mbohlkeschneider) [reply src/gluonts/model/deepvar/_network.py]: Good point. I'll try to make that clearer. I think the confusion stems mostly from not being really relevant to this model - maybe I'll factor this over to the GP model once, which I will work on next.

Comment (mbohlkeschneider) [reply src/gluonts/model/deepvar/_network.py]: I refactored a bit and put some description to the estimator constructor. Hope that is clearer.

File: src/gluonts/model/deepvar/_estimator.py (hunk 1 above)
Comment (mbohlkeschneider) [reply src/gluonts/model/deepvar/_estimator.py]: Not exactly, AFAIK. 

One can use the `to_offset` function from pandas, though, to produce the same thing.

Comment (mbohlkeschneider) [reply src/gluonts/model/deepvar/_estimator.py]: Not exactly. One can use the `to_offset` function from pandas to do the same thing, though.

File: test/model/test_deepvar.py [hunk 3]
```diff
@@ -102,9 +102,9 @@ def test_deepvar(
         freq=metadata.freq,
         distr_output=distr_output,
         scaling=False,
-        use_copula=use_copula,
+        use_marginal_transformation=use_copula,
```
Comment (lcallot) [inline test/model/test_deepvar.py]: you should probably update use_copula here for consistency.

File: test/model/test_deepvar.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def load_multivariate_constant_dataset():
 estimator = DeepVAREstimator
 
 
-@pytest.mark.timeout(20000)
+@pytest.mark.timeout(10)
 @pytest.mark.parametrize(
     "distr_output, num_batches_per_epoch, Estimator, hybridize, use_copula",
```
Comment (lcallot) [inline test/model/test_deepvar.py]: copula -> marginal_transform

File: test/model/test_deepvar.py (hunk 3 above)
Comment (mbohlkeschneider) [reply test/model/test_deepvar.py]: Missed to update that file. Fixed. Thank you!

File: test/model/test_deepvar.py (hunk 4 above)
Comment (mbohlkeschneider) [reply test/model/test_deepvar.py]: See above.

**Pull request merged**